src/main/kotlin/com/atlassian/performance/tools/ssh/api/DetachedProcess.kt

```diff
@@ -11,6 +11,7 @@ import java.util.concurrent.TimeUnit
  *
  * @see [SshConnection.stopProcess]
  */
+@Deprecated(message = "Use BackgroundProcess instead")
 class DetachedProcess private constructor(
```

Comment:
I tried to do it in `DetachedProcess`, but [none of the approaches seemed to satisfy all requirements](https://stash.atlassian.com/projects/JIRASERVER/repos/jira-performance-tests/pull-requests/117/overview?commentId=2752468). So I wrote the tests, which guard the requirements. Maybe someone will come in the future and will be able to make them work with `DetachedProcess`.

But for now, I had to resign from the requirement of starting/stopping on different `Session.Command`s.